frontend/components/markup-modal/index.js

```diff
@@ -2,28 +2,13 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import Modal from 'react-modal'
 import styled from 'styled-components'
-import {
-  badgeUrlFromPath,
-  badgeUrlFromPattern,
-} from '../../../core/badge-urls/make-badge-url'
-import { advertisedStyles } from '../../../supported-features.json'
-import generateAllMarkup from '../../lib/generate-image-markup'
-import { Snippet } from '../snippet'
-import { BaseFont, H3, Badge, BlockInput } from '../common'
+import { BaseFont } from '../common'
 import MarkupModalContent from './markup-modal-content'
 
 const ContentContainer = styled(BaseFont)`
   text-align: center;
 `
 
-const WeeSnippet = ({ snippet, truncate = false }) => (
```

Comment:
I have no idea what I'm looking at here 😄

Reply:
In a nutshell: React is built around a single abstraction called components. Virtually everything in React is a component. The component's `render()` method is the most important function inside it. It gets called every time something changes in the component, and transforms the immutable attributes passed in from the parent – called `props` – and the state that lives in the component – appropriately enough called `state` – into a tree of child components. The leaves of the component tree correspond to DOM elements. After rendering finishes, React intelligently reconciles the component tree with the DOM.

What you're looking at _right_ here is a [function component](https://reactjs.org/docs/components-and-props.html#function-and-class-components); it's basically shorthand where instead of making a class, you just write the render function (it gets a single parameter, the `props`).

In `<Snippet truncate={truncate} fontSize="10pt" snippet={snippet} />`, a `Snippet` component is being created, with three props: `truncate`, `fontSize`, and `snippet`.

These are probably useful starting points for reading React code:

1. https://reactjs.org/docs/rendering-elements.html
2. https://reactjs.org/docs/components-and-props.html
3. https://reactjs.org/docs/state-and-lifecycle.html